frontend/src/pages/Search.tsx

```diff
@@ -60,7 +60,10 @@ export const Search = () => {
             <div className="flex flex-col">
               <div className="flex h-14 justify-between">
                 <h2 className="p-2 text-sm">
-                  Showing {pageIndex * ItemsPerPage + 1}-{(pageIndex + 1) * ItemsPerPage}{" "}
+                  Showing {pageIndex * ItemsPerPage + 1}-
+                  {(pageIndex + 1) * ItemsPerPage < data.count
+                    ? (pageIndex + 1) * ItemsPerPage
+                    : data.count}{" "}
```

Comment:
How about using `Math.min`?
```suggestion
                  {Math.min((pageIndex + 1) * ItemsPerPage, data.count)}{" "}
```